google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITRemoteUDFTest.java

```diff
@@ -0,0 +1,131 @@
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ITRemoteUDFTest {
+
+  private static final String ID = UUID.randomUUID().toString().substring(0, 8);
+  private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
+  private static final String CONNECTION_ID = "test-connectin-id-" + ID;
+  private static final String LOCATION = "US";
+  private static final String PARENT = LocationName.of(PROJECT_ID, LOCATION).toString();
+  private static final String REMOTE_ENDPOINT = "https://aaabbbccc-uc.a.run.app";
+  private static final String ROUTINE_DATASET = RemoteBigQueryHelper.generateDatasetName();
+  private static ConnectionServiceClient client;
+  private static Connection connection;
+  private static BigQuery bigquery;
+
+  @Before
+  public void setUp() throws IOException {
+    RemoteBigQueryHelper bigqueryHelper = RemoteBigQueryHelper.create();
+    bigquery = bigqueryHelper.getOptions().getService();
+    client = ConnectionServiceClient.create();
+
+    DatasetInfo info =
+        DatasetInfo.newBuilder(ROUTINE_DATASET).setDescription("java routine lifecycle").build();
+    bigquery.create(info);
+    CloudResourceProperties cloudResourceProperties = CloudResourceProperties.newBuilder().build();
+    CreateConnectionRequest request =
+        CreateConnectionRequest.newBuilder()
+            .setParent(PARENT)
+            .setConnection(
+                Connection.newBuilder().setCloudResource(cloudResourceProperties).build())
+            .setConnectionId(CONNECTION_ID)
+            .build();
+    connection = client.createConnection(request);
+  }
+
+  @AfterClass
+  public static void afterClass() {
+    if (bigquery != null) {
+      RemoteBigQueryHelper.forceDelete(bigquery, ROUTINE_DATASET);
+    }
+    // delete a connection
+    DeleteConnectionRequest request =
+        DeleteConnectionRequest.newBuilder().setName(connection.getName()).build();
+    client.deleteConnection(request);
+    client.close();
+  }
+
+  @Test
+  public void testRoutineRemoteUDF() {
+    String routineName = RemoteBigQueryHelper.generateRoutineName();
+    RoutineId routineId = RoutineId.of(ROUTINE_DATASET, routineName);
+    Map<String, String> userDefinedContext =
+        new HashMap<String, String>() {
+          {
+            put("key1", "value1");
+            put("key2", "value2");
+          }
+        };
+
+    RemoteFunctionOptions remoteFunctionOptions =
+        RemoteFunctionOptions.newBuilder()
+            .setEndpoint(REMOTE_ENDPOINT)
+            .setConnection(connection.getName())
+            .setMaxBatchingRows(Long.valueOf(30))
+            .setUserDefinedContext(userDefinedContext)
+            .build();
+    RoutineInfo routineInfo =
+        RoutineInfo.newBuilder(routineId)
+            .setRoutineType("SCALAR_FUNCTION")
+            .setBody("x * 3")
```

Comment:
I think the body and language are not needed in the context of a remote function.